home/src/app/theme/components/searchbar/searchbarCtrl.js

```diff
@@ -28,6 +28,14 @@
 				$rootScope.mapPage = true;
 				$rootScope.mapSearchStringTrans = $scope.searchString;
 				$state.go('map', {}, { reload: true });
+			} else {
+				$location.path("/search");
+				if ($location.path() == "/search") {
+					searchService.prepForBroadcast($rootScope.predicates);
+					$scope.$on('handleBroadcast', function() {
```

Comment:
@Zizanzu , It will work without catch broadcast here. You should read the $broadcast angular document for more details